controllers/articles.controller.js

```diff
@@ -7,9 +8,19 @@ const {
 } = require("../models/articles.model.js");
 
 exports.getArticles = (req, res, next) => {
-    selectArticles().then((articles) => {
-        res.status(200).send({ articles });
-    });
+    const { topic } = req.query;
+    selectTopics()
+        .then((topics) => {
+            const validTopics = [];
+            topics.forEach((topic) => {
+                validTopics.push(topic.slug);
+            })
+            return selectArticles(validTopics, topic);
+        })
```

Comment:
Smart :) 
A more efficient way to do this may be to check the specific topic asked for in an SQL query, rather than getting all.